src/performance/monitor.py

```
@@ -0,0 +1,132 @@
+        return metrics
+        
+    def get_history(self) -> List[Dict[str, Any]]:
+        """Return all historical metrics"""
+        return list(self.history)
+        
+    def get_statistics(self) -> Dict[str, Any]:
+        """Calculate statistics from historical data"""
+        if not self.history:
+            return {}
+            
+        operations = set(m["operation"] for m in self.history)
```

Ruff (0.8.2)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)